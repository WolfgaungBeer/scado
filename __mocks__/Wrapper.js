import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
