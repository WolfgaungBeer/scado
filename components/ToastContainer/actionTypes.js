export const SHOW_TOAST = '@SCADO/TOASTCONTAINER/SHOW_TOAST';
export const DISMISS_TOAST = '@SCADO/TOASTCONTAINER/DISMISS_TOAST';
