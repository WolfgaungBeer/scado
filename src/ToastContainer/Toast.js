import React from 'react';
import { string, node, func } from 'prop-types';
import { ToastWrapper, IconWrapper, ComponentWrapper, CloseButtonWrapper, AbsoluteWrapper } from './styled';
import iconMap from './iconMap';
import Icon from '../Icon';

const propTypes = {
    type: string.isRequired,
    content: node.isRequired,
    onClose: func.isRequired,
};

const defaultProps = {};

const Toast = ({ type, content, onClose }) => (
    <ToastWrapper>
        <IconWrapper flex="0 1 15%" justifyContent="center" alignItems="center" color={type}>
            <Icon color="white" scale="l">{iconMap[type]}</Icon>
        </IconWrapper>
        <ComponentWrapper flex="0 1 80%" justifyContent="center" alignItems="center">
            {content}
        </ComponentWrapper>
        <CloseButtonWrapper flex="0 1 5%" justifyContent="center" alignItems="center">
            <AbsoluteWrapper>
                <Icon onClick={onClose}>close</Icon>
            </AbsoluteWrapper>
        </CloseButtonWrapper>
    </ToastWrapper>
);

Toast.propTypes = propTypes;
Toast.defaultProps = defaultProps;

export default Toast;
