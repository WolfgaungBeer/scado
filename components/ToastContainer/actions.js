import uuid from 'uuid/v4';
import { SHOW_TOAST, DISMISS_TOAST } from './actionTypes';
import animationActions from '../Animation/actions';

const showToastAction = payload => ({ type: SHOW_TOAST, payload });

const dismissToastAction = payload => ({ type: DISMISS_TOAST, payload });

const showToast = _payload => (dispatch) => {
    const id = uuid();
    const payload = { ..._payload, id };
    dispatch(animationActions.animate({ id, type: 'enter', className: 'vivify flipInX' }));
    dispatch(showToastAction(payload));
};

const dismissToast = payload => (dispatch) => {
    dispatch(animationActions.animate({ id: payload, type: 'leave', className: 'vivify flipOutX' }));
    setTimeout(() => {
        dispatch(animationActions.remove(payload));
        dispatch(dismissToastAction(payload));
    }, 850);
};

export default { showToast, dismissToast };
