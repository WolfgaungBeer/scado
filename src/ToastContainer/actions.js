import { SHOW_TOAST, DISMISS_TOAST } from './actionTypes';

export const showToast = payload => ({ type: SHOW_TOAST, payload });
export const dismissToast = payload => ({ type: DISMISS_TOAST, payload });

export default { showToast, dismissToast };
