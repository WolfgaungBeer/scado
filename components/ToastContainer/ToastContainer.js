import React from 'react';
import { array, func, string } from 'prop-types';
import { ToastContainerWrapper } from './styled';
import Toast from './Toast';

const propTypes = {
    toasts: array.isRequired,
    dismissToast: func.isRequired,
    position: string
};

const defaultProps = {
    position: 'bottom-right'
};

const ToastContainer = ({ toasts, dismissToast, position }) => {
    const posArray = position.split(/[-]/);

    toasts.forEach((toast) => {
        const { id, dismissTimeout } = toast;
        setTimeout(() => dismissToast(id), dismissTimeout);

    });

    if (!toasts || toasts.length === 0) {
        return null;
    }

    return (
        <ToastContainerWrapper pos1={posArray[0]} pos2={posArray[1]}>
            {toasts.map(t => <Toast key={t.id} type={t.type} component={t.component} onClose={() => dismissToast(t.id)} />)}
        </ToastContainerWrapper>
    );
};

ToastContainer.propTypes = propTypes;
ToastContainer.defaultProps = defaultProps;

export default ToastContainer;
