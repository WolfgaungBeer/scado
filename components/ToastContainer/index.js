import { connect } from 'react-redux';
import { toasts } from './selectors';
import { dismissToast } from './actions';
import ToastContainer from './ToastContainer';

const mapStateToProps = (state) => {
    return {
        toasts: toasts(state)
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        dismissToast: (id) => dispatch(dismissToast(id))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(ToastContainer);
