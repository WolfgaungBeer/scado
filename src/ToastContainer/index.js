import { connect } from 'react-redux';
import { toasts } from './selectors';
import toastActions from './actions';
import ToastContainer from './ToastContainer';

const mapStateToProps = state => ({
    toasts: toasts(state),
});

const mapDispatchToProps = dispatch => ({
    dismissToast: id => dispatch(toastActions.dismissToast(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ToastContainer);
