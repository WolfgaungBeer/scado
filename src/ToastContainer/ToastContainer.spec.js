import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import ToastContainer from './';


it('renders the ToastContainer Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <ToastContainer />
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <ToastContainer position="bottom-left" />
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();
});
