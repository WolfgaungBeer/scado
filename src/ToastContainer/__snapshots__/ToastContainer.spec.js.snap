// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ToastContainer Component correctly 1`] = `
<div
  className="sc-kAzzGY bOLSHM"
>
  <div
    className="sc-chPdSV eMmOGE sc-bwzfXH ksJcPo"
  >
    <div
      className="sc-kgoBCf btUZkU sc-bwzfXH hRHjBy"
      color="success"
    >
      <i
        className="material-icons  sc-ifAKCX cKhEfU"
        color="white"
        onClick={undefined}
        scale="l"
      >
        check_circle
      </i>
    </div>
    <div
      className="sc-kGXeez beVzcQ sc-bwzfXH klpUfB"
    >
      Text
    </div>
    <div
      className="sc-kpOJdX iioVns sc-bwzfXH kDqBmg"
    >
      <div
        className="sc-dxgOiQ itkEVu"
      >
        <i
          className="material-icons  sc-ifAKCX fpvHwA"
          color={undefined}
          onClick={[Function]}
          scale={undefined}
        >
          close
        </i>
      </div>
    </div>
  </div>
  <div
    className="sc-chPdSV eMmOGE sc-bwzfXH ksJcPo"
  >
    <div
      className="sc-kgoBCf jtBGQy sc-bwzfXH hRHjBy"
      color="error"
    >
      <i
        className="material-icons  sc-ifAKCX cKhEfU"
        color="white"
        onClick={undefined}
        scale="l"
      >
        error_outline
      </i>
    </div>
    <div
      className="sc-kGXeez beVzcQ sc-bwzfXH klpUfB"
    >
      Text
    </div>
    <div
      className="sc-kpOJdX iioVns sc-bwzfXH kDqBmg"
    >
      <div
        className="sc-dxgOiQ itkEVu"
      >
        <i
          className="material-icons  sc-ifAKCX fpvHwA"
          color={undefined}
          onClick={[Function]}
          scale={undefined}
        >
          close
        </i>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the ToastContainer Component correctly 2`] = `
<div
  className="sc-kAzzGY iHtRhf"
>
  <div
    className="sc-chPdSV eMmOGE sc-bwzfXH ksJcPo"
  >
    <div
      className="sc-kgoBCf btUZkU sc-bwzfXH hRHjBy"
      color="success"
    >
      <i
        className="material-icons  sc-ifAKCX cKhEfU"
        color="white"
        onClick={undefined}
        scale="l"
      >
        check_circle
      </i>
    </div>
    <div
      className="sc-kGXeez beVzcQ sc-bwzfXH klpUfB"
    >
      Text
    </div>
    <div
      className="sc-kpOJdX iioVns sc-bwzfXH kDqBmg"
    >
      <div
        className="sc-dxgOiQ itkEVu"
      >
        <i
          className="material-icons  sc-ifAKCX fpvHwA"
          color={undefined}
          onClick={[Function]}
          scale={undefined}
        >
          close
        </i>
      </div>
    </div>
  </div>
  <div
    className="sc-chPdSV eMmOGE sc-bwzfXH ksJcPo"
  >
    <div
      className="sc-kgoBCf jtBGQy sc-bwzfXH hRHjBy"
      color="error"
    >
      <i
        className="material-icons  sc-ifAKCX cKhEfU"
        color="white"
        onClick={undefined}
        scale="l"
      >
        error_outline
      </i>
    </div>
    <div
      className="sc-kGXeez beVzcQ sc-bwzfXH klpUfB"
    >
      Text
    </div>
    <div
      className="sc-kpOJdX iioVns sc-bwzfXH kDqBmg"
    >
      <div
        className="sc-dxgOiQ itkEVu"
      >
        <i
          className="material-icons  sc-ifAKCX fpvHwA"
          color={undefined}
          onClick={[Function]}
          scale={undefined}
        >
          close
        </i>
      </div>
    </div>
  </div>
</div>
`;
